specs/Standards/Tokens/FungibleTokenMetadata.md

```diff
@@ -0,0 +1,99 @@
+# Fungible Token Metadata ([NEP-148](https://github.com/near/NEPs/discussions/148))
+
+Version `1.0.0`
+
+## Summary
+[summary]: #summary
+
+An interface for a fungible token's metadata. The goal is to keep the metadata future-proof as well as lightweight. This will be important to dApps needing additional information about an FT's properties, and broadly compatible with other tokens standards such that the [NEAR Rainbow Bridge](https://near.org/blog/eth-near-rainbow-bridge/) can move tokens between chains.
+
+## Motivation
+
+Custom fungible tokens play a major role in decentralized applications today. FTs can contain custom properties to differentiate themselves from other tokens or contracts in the ecosystem. In NEAR, many common properties can be stored right on-chain. Other properties are best stored off-chain or in a decentralized storage platform, in order to save on storage costs and allow rapid community experimentation.
+
+As blockchain technology advances, it becomes increasingly important to provide backwards compatibility and a concept of a spec. This standard encompasses all of these concerns.
+
+Prior art:
+
+- [EIP-1046](https://eips.ethereum.org/EIPS/eip-1046)
+- [OpenZeppelin's ERC-721 Metadata standard](https://docs.openzeppelin.com/contracts/2.x/api/token/erc721#ERC721Metadata) also helped, although it's for non-fungible tokens.
+
+## Guide-level explanation
+
+A fungible token smart contract allows for discoverable properties. Some properties can be determined by other contracts on-chain, or return in view method calls. Others can only be determined by an oracle system to be used on-chain, or by a frontend with the ability to access a linked reference file.
+
+### Examples scenario
+
+#### Token provides metadata upon deploy and initialization
+
+Alice deploys a wBTC fungible token contract.
+
+**Assumptions**
+
+- The wBTC token contract is `wbtc`.
+- Alice's account is `alice`.
+- The precision ("decimals" in this metadata standard) on wBTC contract is `10^8`.
+
+**High-level explanation**
+
+Alice issues a transaction to deploy and initialize the fungible token contract, providing arguments to the initialization function that set metadata fields.
+
+**Technical calls**
+
+1. `alice` deploys a contract and calls `wbtc::new({"owner_id": "wbtc", "total_supply": "100000000000000", "spec": "ft-1.0.0", "name": "Wrapped Bitcoin", "symbol": "WBTC", "reference": "https://example.com/wbtc.json", "reference_hash": "7c879fa7b49901d0ecc6ff5d64d7f673da5e4a5eb52a8d50a214175760d8919a", "decimals": 8})`.
```

Comment:
reference hash has to be `base64` of 32 bytes, not hex. E.g. `AK3YRHqKhCJNmKfV6SrutnlWW/icN5J8NUPtKsNXR1M=`

Reply:
Resolved in 6bea880bac30ffc9180f780cc11033609d5a0be1